test/jdk/javax/net/ssl/SSLSession/TestEnabledProtocols.java

```diff
@@ -149,6 +147,17 @@ protected void runClientApplication(SSLSocket sslSocket) throws Exception {
         }
     }
 
+    private boolean isConnectionReset(SSLException ssle) {
+        Throwable cause = ssle.getCause();
+        if (cause instanceof SocketException
+                && "Connection reset".equals(cause.getMessage())) {
+            System.out.println("Client SSLException:");
+            ssle.printStackTrace(System.out);
```

Comment:
Sorry for nit picking - but I would keep that in the catch close above. 

```
        } catch (SSLException ssle) {
            if (isConnectionReset(ssle)) {
                System.out.println("Client SSLException:");
                ssle.printStackTrace(System.out);
            } else {
                 failTest(ssle, "Client got UNEXPECTED SSLException:");
            }
        }
 ```

Reply:
That is perfectly fine. It does make it a bit more legible.